### PR TITLE
HSEARCH-2035 Avoiding dependency to Lucene's Query type in TimeoutMan…

### PR DESCRIPTION
**Pull request by gunnarmorling**
…agerImpl

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/query/engine/impl/HSQueryImpl.java [hunk 1]
```diff
@@ -238,7 +238,7 @@ private TimeoutManagerImpl getTimeoutManagerImpl() {
 			if ( luceneQuery == null ) {
 				throw new AssertionFailure( "Requesting TimeoutManager before setting luceneQuery()" );
 			}
-			timeoutManager = new TimeoutManagerImpl( luceneQuery, timeoutExceptionFactory, this.extendedIntegrator.getTimingSource() );
+			timeoutManager = new TimeoutManagerImpl( String.valueOf( luceneQuery ), timeoutExceptionFactory, this.extendedIntegrator.getTimingSource() );
```
Comment (Sanne) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/HSQueryImpl.java]: I think we should avoid invoking the toString() this early, as the code in many Query implementations is a bit stressing.

Could we make it lazy? Worst case just accepting the 'Object' type would do, and we invoke toString() on that as necessary.. but maybe you have a more elegant idea which doesn't take you more than 5 minutes?

Comment (Sanne) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/HSQueryImpl.java]: BTW the reasoning is that the string is only needed when and ever if the timeout exception is thrown.

Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/HSQueryImpl.java]: `Object` it is then. Rebased and force-pushed an update.

**Comment by Sanne** [comment]
merged

**Pull request merged**